CMakeLists.txt

```diff
@@ -1,5 +1,19 @@
 cmake_minimum_required(VERSION 3.1)
 
+# Get list of variables provided via command line arguments.
+# MUST be done before call to 'project'.
+# Via https://cmake.org/pipermail/cmake/2018-January/067002.html
+if(NOT AWS_CMAKE_CL_ARGS)
```

Comment:
should this be moved to `aws_prebuild_dependency`?

Reply:
Gathering cmd args cannot be a part of `aws_prebuild_dependency`, as it should be performed at the very beginning of cmake configuration step. I've extracted this block to a separate function instead: https://github.com/awslabs/aws-c-common/pull/1161